Various improvements to event handling

FichteFoll opened the pull request:
I did this as a follow-up to #115 primarily and then addressed a few things that caught my eye.

Note that these changes are untested!

FichteFoll committed:
Exclude icons folder from archives

This decreases the package size since these icons aren't used by the
package but need to be uploaded separately as assets to Discord.

FichteFoll committed:
Only update activity when changing the active view

Also remove some dead code in `handle_activity` and simplify `is_view_active` for ST4.

FichteFoll committed:
Some cleanups, mostly linter warnings

FichteFoll committed:
Reset activity after leaving ST for a configurable amount of time

Snazzah reviewed (changes requested):
There are some issues with this. I'm using the same functionality as [VSCode Discord Presence](https://github.com/iCrawl/discord-vscode) in terms of how the status is displayed, since it just shows the currently viewed file. An idle timeout is a cool addition but should default just show what the user is actively viewing, rather than use timeouts. (Yes, "idle" in this case doesn't really mean idle but is just a default activity, its misleading in hindsight, but i'd rather keep existing functionality as default)

Some other issues:
- When refocusing the window, it does not show the file anymore
- The idle status flickers when switching between files
- I'd rather keep the post save call, since there are template variables for size and LOC.

FichteFoll committed:
Default to no idle timeout (disabling the feature)

Also fix flickering caused by forgetting to delete a line.

FichteFoll committed:
Re-add post-save handler to refresh template variables

FichteFoll committed:
Opt-in to Python 3.8

FichteFoll commented:
Good points, I've addressed them.


>  When refocusing the window, it does not show the file anymore

I can't really see how this could happen (and didn't attempt to reproduce it either since I don't use Discord rich presence at all and would need to enable it in settings first, yadda yadda).

I also added the file to opt into Python 3.8 since I think it should Just Work, but just as before I did not actually test it.

Snazzah committed:
Fix window re-entering
